sympy/core/function.py

```diff
@@ -566,7 +570,10 @@ def _get_mpmath_func(fname):
         # XXX + 5 is a guess, it is similar to what is used in evalf.py. Should
         #     we be more intelligent about it?
         try:
-            args = [arg._to_mpmath(prec + 5) for arg in self.args]
+            if _eval_mpmath is None:
+                args = [arg._to_mpmath(prec + 5) for arg in self.args]
+            else:
+                args = [arg._to_mpmath(prec + 5) for arg in args]
```

Comment:
It would be better to assign `args` in the `if/else` above.

Reply:
I thought of the same, but this is a try-except block. I would have to add try-except on the top.
https://github.com/sympy/sympy/blob/38b383368f329fbdf43961c68807f35c9d4964dc/sympy/core/function.py#L572-L576

Reply:
Can you not just do `args = self.args`?

Reply:
Yes, that is better